libpod/container_api.go

```diff
@@ -582,7 +597,15 @@ func (c *Container) WaitForExit(ctx context.Context, pollInterval time.Duration)
 		case <-ctx.Done():
 			return -1, fmt.Errorf("waiting for exit code of container %s canceled", id)
 		default:
-			time.Sleep(pollInterval)
+			if conmonPidFd != -1 && !conmonPidFdTriggered {
+				// If possible (pidfd works), the first cycle we block until conmon dies
+				// If this happens, and we fall back to the old poll delay
+				fds := []unix.PollFd{{Fd: int32(conmonPidFd), Events: unix.POLLIN}}
+				_, _ = unix.Poll(fds, -1)
```

Comment:
should we handle the error here?

Reply:
I was thinking of possibly handling EINTR. But, I avoided it, because in general any kind of "smart handling" may get things wrong and accidentally turn this into a busy wait loop. Better to just fall back to the old "safe" timeout loop in case anything weird happens.